core/src/main/scala/akka/persistence/r2dbc/journal/PersistenceReprSerializer.scala

```diff
@@ -23,52 +23,54 @@ import scala.collection.immutable.Set
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Try
 
+final private[akka] class PersistenceReprSerializer(serialization: Serialization)(implicit ec: ExecutionContext) {
```

Comment:
That is fine. However a serializer does not only serialize otherwise it is like a hash. It is like writing a `Cryptography` class and calling it EncryptDecrypt.

Reply:
As long as we want to follow akka persistence there is no such a thing called SerDe it is all serializer.

Reply:
As long as we want to follow akka persistence I think we should serializer.

Reply:
I have to disagree. I believe names should provide as much context as possible. 

Firstly, this is an implementation detail (that's not exposed to the user) and we're not implementing any Akka trait, so there's nothing to follow here. Though we can choose to align with industry conventions, which I believe is what I've done here (more on that below).

Secondly, I ascribe to the philosophy of writing "programs that do one thing and do it well". I firmly believe that the _only_ role a serializer has is to transform data from one format to another; admittedly, in this case it transforms data in both directions, hence why it's called a "serde". I am not sure I understand what you mean by "it is like a hash" in your argument, can you elaborate?

Thirdly, I think it's pretty standard these days to name objects that handle data both ways `SerDe`s. [Kafka](https://spark.apache.org/docs/2.1.0/api/java/org/apache/spark/api/r/SerDe.html), [Spark](https://spark.apache.org/docs/2.1.0/api/java/org/apache/spark/api/r/SerDe.html), [Hadoop Hive](https://hive.apache.org/javadocs/r1.2.2/api/org/apache/hadoop/hive/serde2/SerDe.html), to name but a few all do that. And plenty more do.

Is it a naming convention that's any better than using the other established term, Serializer? I personally think it is, considering how many data frameworks have both `Serializer` and `Deserializer` interfaces, I believe it clarifies matters.

Finally, I think the `Cryptography` example you give is a rather contrived one. Playing along, if  `Cryptography` is a trait that offers `encrypt`/`decrypt` methods I would name `Cryptography`, `Algorithm` or something to that effect . And for each algorithm you'd have an aptly named class that implements the trait.  However, as I mention above, using separate interfaces for (what could be complex) bi-directional operations is a sound and quite common pattern. So, if I was using a single class that implements both the `Encrypt` and `Decrypt` interfaces, then yes I would name it `WhateverEncDec`, so no one has to look at its code to know it implements both.